app/client/components/Upload.tsx

```diff
@@ -98,11 +111,17 @@ export class Upload extends React.Component<{}, UploadState> {
         }
     }
 
-    private handleSubmit() {
+    private async handleSubmit() {
         // TODO: construct the Video object, and write it to the network.
         // This should probably also place the video in the user registry.
-        console.log("TODO actually upload the video!");
-        this.form.submit();
+        const video: Video =
+            await Video.new(this.state.videoTitle, this.state.videoDescription, this.state.videoFile);
+        withDropP(await video.xorName(), async (n) => {
+            const xorName: SerializedDataID = await n.serialise();
+            // TODO: stuff the link in the user profile
+            console.log(`uploaded video to: frames://${xorName.toString("base64")}`);
+        });
+        this.context.router.history.push("/discover");
```

Comment:
How about redirecting the user to their new video page instead of discover?

(This doesn't have to be changed now; just a thought for once the watch page and upload capabilities are more fleshed out.)